lib/datadog/profiling/component.rb

```diff
@@ -84,13 +84,24 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
       end
 
       private_class_method def self.build_thread_context_collector(settings, recorder, optional_tracer, timeline_enabled)
+        otel_context_enabled =
+          case settings.profiling.advanced.preview_otel_context_enabled
+          when 'both'
+            :both
+          when 'only'
+            :only
+          else
+            false
+          end
```

Comment:
Q: I was wondering if this was a style choice or if there's some limitation of the settings system that prevents us from using `:both`, `:only` and `false` directly as the settings normalized values?

Reply:
Fixed in 222b4d4b9ea60e48c6ad8319e0aa85cf5a747b31 (PR #3987). To be honest I wasn't sure about some of the details in the settings system, but on a second pass it works quite nicely.